### ISSUE TITLE
[Android][Build 2.1.5-3] Query test "select" failed with float number

### ISSUE THREAD

**Issue by euniceatcouchbase** [couchbase/couchbase-lite-android#1915]
Functional test case "test_getDoc_withValueTypeDouble[doc_with_double_1]" failed while selecting a float number from the db.

Repro Steps:
1. unzip and use the attached cblite2 database file, 
[db_name.cblite2.zip](https://github.com/couchbase/couchbase-lite-android/files/3159761/db_name.cblite2.zip)
2. use cblite tool to select item_price == 2.754 

Expected result:
return 1 row of data which has item_price is 2.754

Actual result:
return 0 row of data

Additional info (see attached logs):
[cblite_tool_results.txt](https://github.com/couchbase/couchbase-lite-android/files/3159769/cblite_tool_results.txt)

./cblite query ~/temp/db_name.cblite2/  '{WHAT: [["."]]}' 
[ * ]
-----
[{"item":3,"item_price":3.754}]
**[{"item":2,"item_price":2.754}]**
[{"item":1,"item_price":1.754}]
[{"item":0,"item_price":0.754}]
[{"item":4,"item_price":4.754}]

./cblite query ~/temp/db_name.cblite2/  '{WHAT: [["."]],WHERE:["=",[".","**item_price**"],2.754]}'
[ * ]
-----

./cblite query  ~/temp/db_name.cblite2/  '{WHAT: [["."]],WHERE:["=",[".","**item**"],2]}'
[ * ]
-----
[{"item":2,"item_price":2.754}]

The following is from Android TestServer while search_query.execute();

SELECT fl_result(key), fl_result(fl_value(body, 'item')), fl_result(fl_value(body, 'item_price')) FROM kv_default WHERE (fl_value(body, 'item_price') = 2.754) AND (flags & 1) = 0

**Comment by snej** [comment]
The number `2.754` can't be represented exactly in binary floating point, so there will be roundoff error. This can result in equality tests failing, depending on how the number got rounded by various pieces of code during ASCII–binary conversions. In particular, converting "2.754" to a float isn't going to give the same result as converting it to a double, so comparing the two types will fail.

tl;dr: This test is probably invalid and shouldn't be used. If you want to test FP for equality, you have to add a fudge factor, like `abs(x - y) < 0.0001`.

**Comment by bmeike** [comment]
I think I'm gonna make like a lawyer and counter-sue.  I'm gonna open an issue against the test!

**Comment by bmeike** [comment]
As described, this isn't a bug.  If there is still a problem after accounting for roundoff error, please open a new issue.

**Comment by euniceatcouchbase** [comment]
Tested with a test framework fix from Sridevi's working-in-progress branch, which handles this issue.
